Sources/Algorithms/FirstNonNil.swift

```diff
@@ -0,0 +1,28 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+//===----------------------------------------------------------------------===//
+// firstNonNil(_:)
+//===----------------------------------------------------------------------===//
+
+public extension Sequence {
+    /// Returns the first element in `self` that `transform` maps to a `.some`.
```

Comment:
This method doesn't return the first element, it returns the mapped result. Can you update this wording and include an example demonstrating the usage?